commit 1789c57dc97ea2f9819ef89e28bf17208b6208e7
Author: Florian Weimer <fweimer@redhat.com>
Date:   Fri May 2 17:41:43 2025 +0200

    libsanitizer: Fix build with glibc 2.42
    
    The termio structure will be removed from glibc 2.42.  It has
    been deprecated since the late 80s/early 90s.
    
    Cherry-picked from LLVM commit 59978b21ad9c65276ee8e14f26759691b8a65763
    ("[sanitizer_common] Remove interceptors for deprecated struct termio
    (#137403)").
    
    Co-Authored-By: Tom Stellard <tstellar@redhat.com>
    
    libsanitizer/
    
            * sanitizer_common/sanitizer_common_interceptors_ioctl.inc: Cherry
            picked from LLVM commit 59978b21ad9c65276ee8e14f26759691b8a65763.
            * sanitizer_common/sanitizer_platform_limits_posix.cpp: Likewise.
            * sanitizer_common/sanitizer_platform_limits_posix.h: Likewise.

Index: gcc-13.4.0+git9739/libsanitizer/sanitizer_common/sanitizer_common_interceptors_ioctl.inc
===================================================================
--- gcc-13.4.0+git9739.orig/libsanitizer/sanitizer_common/sanitizer_common_interceptors_ioctl.inc	2025-06-05 18:03:58.000000000 +0200
+++ gcc-13.4.0+git9739/libsanitizer/sanitizer_common/sanitizer_common_interceptors_ioctl.inc	2025-10-09 16:28:01.433128974 +0200
@@ -338,17 +338,9 @@ static void ioctl_table_fill() {
   _(SOUND_PCM_WRITE_CHANNELS, WRITE, sizeof(int));
   _(SOUND_PCM_WRITE_FILTER, WRITE, sizeof(int));
   _(TCFLSH, NONE, 0);
-#if SANITIZER_GLIBC
-  _(TCGETA, WRITE, struct_termio_sz);
-#endif
   _(TCGETS, WRITE, struct_termios_sz);
   _(TCSBRK, NONE, 0);
   _(TCSBRKP, NONE, 0);
-#if SANITIZER_GLIBC
-  _(TCSETA, READ, struct_termio_sz);
-  _(TCSETAF, READ, struct_termio_sz);
-  _(TCSETAW, READ, struct_termio_sz);
-#endif
   _(TCSETS, READ, struct_termios_sz);
   _(TCSETSF, READ, struct_termios_sz);
   _(TCSETSW, READ, struct_termios_sz);
Index: gcc-13.4.0+git9739/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp
===================================================================
--- gcc-13.4.0+git9739.orig/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp	2025-10-09 16:28:01.409128564 +0200
+++ gcc-13.4.0+git9739/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cpp	2025-10-09 16:28:01.433128974 +0200
@@ -459,9 +459,6 @@ unsigned struct_ElfW_Phdr_sz = sizeof(El
   unsigned struct_input_id_sz = sizeof(struct input_id);
   unsigned struct_mtpos_sz = sizeof(struct mtpos);
   unsigned struct_rtentry_sz = sizeof(struct rtentry);
-#if SANITIZER_GLIBC || SANITIZER_ANDROID
-  unsigned struct_termio_sz = sizeof(struct termio);
-#endif
   unsigned struct_vt_consize_sz = sizeof(struct vt_consize);
   unsigned struct_vt_sizes_sz = sizeof(struct vt_sizes);
   unsigned struct_vt_stat_sz = sizeof(struct vt_stat);
Index: gcc-13.4.0+git9739/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
===================================================================
--- gcc-13.4.0+git9739.orig/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h	2025-10-09 16:28:01.409128564 +0200
+++ gcc-13.4.0+git9739/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h	2025-10-09 16:28:01.433128974 +0200
@@ -997,7 +997,6 @@ extern unsigned struct_hd_geometry_sz;
 extern unsigned struct_input_absinfo_sz;
 extern unsigned struct_input_id_sz;
 extern unsigned struct_mtpos_sz;
-extern unsigned struct_termio_sz;
 extern unsigned struct_vt_consize_sz;
 extern unsigned struct_vt_sizes_sz;
 extern unsigned struct_vt_stat_sz;
